ci-operator/step-registry/assisted/agent/test/assisted-agent-test-ref.yaml

```
@@ -0,0 +1,15 @@
+ref:
+  as: assisted-agent-test
+  from: src
```

usually if the step is specific to some code, then it should be using the specific image
e.g.:
```suggestion
  from: assisted-installer-agent-code
```

And in ``openshift-assisted-installer-agent-master.yaml``:
```
images:
...
- from: src
  to: assisted-installer-agent-code
```

---

same goes to the other steps

---

is there a specific reason to have such indirection or it's just a convention?

---

the reason for that indirection is to make it runnable on other repos. ci-operator is built to steps being reused anywhere needed even when it's for jobs of other repos

in this specific case, the major advantage is actually explicitly not allow using it for ``src`` that's originating from other repos. the main scenarios of just using ``src`` in steps is e.g. general step to lint code and such thing

---

for a completely generic image, I think there's ``cli`` as well

---

And also, it makes sense, thanks for the explanation